### PR TITLE
sales and listing analytics added

### PR DESCRIPTION
**Pull request by MalikSaqibIftikhar**
## Description

Fixes #

## Checklist
- [x] Targeted PR against correct branch.
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Run `meteor npm run lint`
- [ ] Added an entry to the `CHANGELOG.md` file.
- [ ] Re-reviewed `Files changed` in the Github PR explorer.

### CODE REVIEW

File: imports/api/listings/server/methods.js [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+                    time: txns[i]?.tx_response?.timestamp
+                }
+
+                // inserting the extracted information in sales collection
+                Listings.upsert({'txhash': txns[i].txhash},{$set: listing})
+
+            }
+            
+        }
+        catch (e) {
```
Comment (mustafapylons) [inline imports/api/listings/server/methods.js]: There are multiple catch (error) empty blocks. It's not a right way to handle exceptions.

File: imports/api/listings/server/methods.js [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+//fetching username info 
+function getUserNameInfo(address){
+    var result;
+    try{
+        let response = HTTP.get(
+            sanitizeUrl(`${Meteor.settings.remote.api}/pylons/account/address/${address}`)
+        );
+        result = JSON.parse(response.content);
+    }
+    catch(e){
```
Comment (mustafapylons) [inline imports/api/listings/server/methods.js]: Same empty catch error block here.

File: imports/api/sales/server/methods.js [hunk 3]
```diff
@@ -0,0 +1,281 @@
...
+//fetching the username info
+function getUserNameInfo(address){
+    var result;
+    try{
+        let response = HTTP.get(
+            sanitizeUrl(`${Meteor.settings.remote.api}/pylons/account/address/${address}`)
+        );
+        result = JSON.parse(response.content);
+    }
+    catch(e){
```
Comment (mustafapylons) [inline imports/api/sales/server/methods.js]: same here with try, catch

**Review by mustafapylons — COMMENTED** [review]
Exception handling has to be looked over before approval.

**Review by mustafapylons — APPROVED** [review]
Looks good to me now.

**Comment by MalikSaqibIftikhar** [comment]
Will be committed again with combined server and client code

**Pull request closed without merging**